src/test/groovy/graphql/util/javac/DynamicJavacSupport.java

```diff
@@ -0,0 +1,157 @@
+package graphql.util.javac;
+
+import javax.tools.DiagnosticCollector;
+import javax.tools.FileObject;
+import javax.tools.ForwardingJavaFileManager;
+import javax.tools.JavaCompiler;
+import javax.tools.JavaFileManager;
+import javax.tools.JavaFileObject;
+import javax.tools.SimpleJavaFileObject;
+import javax.tools.StandardJavaFileManager;
+import javax.tools.ToolProvider;
+import java.io.ByteArrayOutputStream;
+import java.io.OutputStream;
+import java.net.URI;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static java.util.Objects.requireNonNull;
+
+
+/**
+ * This utility allows is to dynamically create Java classes and place them into
+ * floating class loaders.  This will allow us to test class loader challenges
+ * <p>
+ * Proprs to https://www.baeldung.com/java-string-compile-execute-code where
+ * most of this code came from.
+ */
+public class DynamicJavacSupport {
```

Comment:
I added this to allow is to generate classes in the tests - that are in seperate and diverse class loaders.

Neato!